lib/clearwater/component.rb

```diff
@@ -27,12 +27,12 @@ def tag tag_name, attributes=nil, content=nil
     end
 
     def to_s
-      html = render.to_s
-      if html.respond_to? :html_safe
-        html = html.html_safe
+      content = Array(render).map do |node|
+        html = node.to_s
+        html.respond_to?(:html_safe) ? html.html_safe : html
       end
 
-      html
+      content.join
```

Comment:
Good catch; it returns a new string which would not be flagged as safe. I'll edit it to just call `html_safe` after the array is joined.